src/main/java/Deadline.java

```diff
@@ -0,0 +1,26 @@
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+public class Deadline extends Task implements Serializable {
+
+    protected String by;
+    LocalDate date;
+
+    public Deadline(String description, String by) {
```

Comment:
Perhaps you can write Javadoc comments for public methods? It is stated in the [style guide](https://se-education.org/guides/conventions/java/intermediate.html#comments).
I noticed this in several other places too.